value/src/test/java/com/google/auto/value/processor/ExtensionTest.java

```diff
@@ -120,12 +121,49 @@ public void testExtensionConsumesProperties() throws Exception {
     );
     assertAbout(javaSource())
         .that(javaFileObject)
-        .processedWith(
-            new AutoValueProcessor(ImmutableList.<AutoValueExtension>of(new FooExtension())))
+        .processedWith(new AutoValueProcessor(ImmutableList.of(new FooExtension())))
         .compilesWithoutError()
         .and().generatesSources(expectedExtensionOutput);
   }
 
+  public void testDoesntRaiseWarningOnConsumedProperties() {
+    JavaFileObject impl = JavaFileObjects.forSourceLines("foo.bar.Baz",
+        "package foo.bar;",
+        "import com.google.auto.value.AutoValue;",
+        "@AutoValue public abstract class Baz {",
+        "  abstract String foo();",
+        "  abstract String dizzle();",
+        "}");
+    assertAbout(javaSource())
+        .that(impl)
+        .processedWith(new AutoValueProcessor(ImmutableList.of(new FooExtension())))
+        .compilesWithoutError()
+        // expected warnings: @AutoValue and @Generated annotations not consumed
+        .withWarningCount(2);
```

Comment:
I would expect that would pass since you're suppressing all processor warnings.  The point of this test is to ensure that those two expected warnings, **and no others**, are thrown.  In the error case, a warning is thrown for every property, but `.withCompilerOptions("-Xlint:-processing")` would suppress those.